server/einterfaces/pap.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package einterfaces
+
+import (
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/request"
+)
+
+// PolicyAdministrationPointInterface is the service that manages access control policies.
+// It is responsible for creating, updating, and deleting policies.
+// Also, it provides methods to check the validity of expressions and to retrieve policies.
+type PolicyAdministrationPointInterface interface {
+	// Init initializes the policy administration point and intiates the CEL engine.
+	// It is an idempotent operation, meaning that it can be called multiple times.
+	Init(rctx request.CTX) *model.AppError
+	// BasicAutocomplete returns a map of basic autocomplete values for the given target type.
+	GetBasicAutocomplete(rctx request.CTX, targetType string) (map[string]any, *model.AppError)
+	// CheckExpression checks the validity of the given expression using the CEL engine.
+	// It returns a list of CELExpressionError if the expression is invalid.
+	// If the expression is valid, it returns an empty list.
+	CheckExpression(rctx request.CTX, expression string) ([]model.CELExpressionError, *model.AppError)
```

Comment:
Not familiar with the interface, but when would we want to `CheckExpression` without also doing `ExtractAttributeFields`? Would it make sense to consolidate the API into one method that returns errors or the fields? (0/5, just thinking out loud)

Reply:
We can do that, while we are relying on the `cel-go` library to check expression, extracting attributes is a bit more tricky that we are doing it by ourselves. We can combine those for sure but for checking the expression we don't necessarily think about attributes. Consolidation would make sense, but I'd prefer doing that later as this is not the public API.